wp-includes/template.php

```diff
@@ -171,3 +171,57 @@ function wp_service_worker_error_details_template( $output = '' ) {
 function wp_service_worker_error_message_placeholder() {
 	echo '<p><!--WP_SERVICE_WORKER_ERROR_MESSAGE--></p>';
 }
+
+/**
+ * Reload the offline page and check if user comes online.
+ *
+ * @since 0.7
+ */
+function wp_service_worker_offline_page_reload() {
+	if ( isset( $_SERVER['REQUEST_METHOD'] ) && 'GET' !== $_SERVER['REQUEST_METHOD'] ) {
+		return;
+	}
+
+	if ( ! is_offline() && ! is_500() ) {
+		return;
+	}
+
+	ob_start();
+	?>
+	<script type="module">
+		/**
+		 * Listen to changes in the network state, reload when online.
+		 * This handles the case when the device is completely offline.
+		 */
+		window.addEventListener('online', () => {
+			window.location.reload();
+		});
+
+		// Create a counter to implement exponential backoff.
+		let count = 0;
+
+		/**
+		 * Check if the server is responding and reload the page if it is.
+		 * This handles the case when the device is online, but the server is offline or misbehaving.
+		 */
+		async function checkNetworkAndReload() {
+			try {
+				const response = await fetch(location.href, {method: 'HEAD'});
+				// Verify we get a valid response from the server
+				if (response.status >= 200 && response.status < 500) {
+					window.location.reload();
+					return;
+				}
+			} catch {
+				// Unable to connect so do nothing.
+			}
+			window.setTimeout(checkNetworkAndReload, Math.pow(2, count++) * 2500);
+		}
+		checkNetworkAndReload();
+	</script>
+	<?php
+	return ob_get_clean();
```

Comment:
Why adding the return here? If I'm not mistaken, the output buffering is causing the `script` to no longer be printed onto the page? The output buffering should rather be done in the tests.

Reply:
> output buffering is causing the script to no longer be printed

@westonruter Ahh. my bad. Might get confused during the testing. 
Fixed this now 👍